### PR TITLE
Feature(Wallet): Added packets

### PR DESCRIPTION
**Pull request by angelotadeucci**
Added packet for HaviFruit, Rue, Treva and Valor;
Added commands for testing purposes.

### CODE REVIEW

File: MapleServer2/Packets/WalletPacket.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+                    break;
+                default:
+                    break;
+            }
+
+            PacketWriter pWriter = PacketWriter.Of(SendOp.MONEY_TOKEN);
+
+            pWriter.WriteByte(currency); // type of currency
+            pWriter.WriteLong(amount); // currency amount
+            pWriter.Write("FF FF FF FF FF FF FF FF".ToByteArray()); // always the same
```
Comment (newgithubaccountlol) [inline MapleServer2/Packets/WalletPacket.cs:34]: WriteLong(-1)

File: MapleServer2/Packets/WalletPacket.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+{
+    class WalletPacket
+    {
+        public static Packet UpdateWallet(CurrencyType type, long amount)
+        {
+            byte currency = 0;
+            switch (type)
+            {
+                case CurrencyType.ValorToken:
+                    currency = 3;
```
Comment (AlanMorel) [inline MapleServer2/Packets/WalletPacket.cs:15]: Is it possible to just give these currencies an enum value? That should let you not need this switch case at all.

File: MapleServer2/Tools/GameCommandActions.cs [hunk 3]
```diff
@@ -14,9 +14,21 @@ public static class GameCommandActions
     {
         public static void Process(GameSession session, string command)
         {
-            string[] args = command.ToLower().Split(" ", 2);
+            string[] args = command.ToLower().Split(" ");
             switch (args[0])
             {
+                case "setvalor":
+                    session.Player.Wallet.ValorToken.SetAmount(ParseLong(session, args.Length > 1 ? args[1] : ""));
```
Comment (newgithubaccountlol) [inline MapleServer2/Tools/GameCommandActions.cs:21]: this code is repeated many times. it seems like it's doing it for all cases so you can just move it out.

**Review by AlanMorel — CHANGES_REQUESTED** [review]

File: MapleServer2/Packets/WalletPacket.cs (hunk 2 above)
Comment (newgithubaccountlol) [inline MapleServer2/Packets/WalletPacket.cs:15]: do we want to add currency type to the enum?
https://www.tutorialsteacher.com/csharp/csharp-enum

**Review by newgithubaccountlol — CHANGES_REQUESTED** [review]

File: MapleServer2/Packets/WalletPacket.cs (hunk 2 above)
Comment (angelotadeucci) [reply MapleServer2/Packets/WalletPacket.cs:15]: Sure!

File: MapleServer2/Tools/GameCommandActions.cs (hunk 3 above)
Comment (angelotadeucci) [reply MapleServer2/Tools/GameCommandActions.cs:21]: Sorry, you mean remove the command?

Comment (AlanMorel) [reply MapleServer2/Tools/GameCommandActions.cs:21]: You can't move it out because it's calling it on a different currency each case

**Review by AlanMorel — CHANGES_REQUESTED** [review]

File: MapleServer2/Tools/GameCommandActions.cs (hunk 3 above)
Comment (newgithubaccountlol) [reply MapleServer2/Tools/GameCommandActions.cs:21]: long amount = ParseLong(session, args.Length > 1 ? args[1] : ""));
...
...SetAmount(amount);

Comment (newgithubaccountlol) [reply MapleServer2/Tools/GameCommandActions.cs:21]: Never mind, I'm wrong about how these commands work.

**Pull request merged**